client/homepage/layout.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+import { Component, createRef } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+class Layout extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			inboxHeight: this.getInboxHeight(),
+			showInbox: true,
+		};
+		this.inbox = createRef();
+
+		this.handleScroll = this.handleScroll.bind( this );
+	}
+
+	getInboxHeight() {
+		if ( window.innerWidth <= 782 ) {
+			return 'auto';
+		}
+		const scroll = window.scrollY;
+		const base = 90;
+		return `calc(100vh ${ base - scroll < 0 ? '+' : '-' } ${ Math.abs(
+			base - scroll
+		) }px)`;
+	}
+
+	componentDidMount() {
+		window.addEventListener( 'scroll', this.handleScroll );
+		this.inbox.current.addEventListener( 'scroll', this.handleInboxScroll );
```

Comment:
Pretty edge-case scenario, but on resize this could create some height issues.  We could add a resize listener here as well to cover this.

<img width="706" alt="Screen Shot 2020-05-11 at 3 27 12 PM" src="https://user-images.githubusercontent.com/10561050/81561727-4d57c380-939c-11ea-9714-23dcfc5d7d74.png">